packages/flutter/lib/src/rendering/sliver_persistent_header.dart

```diff
@@ -595,12 +601,18 @@ abstract class RenderSliverFloatingPersistentHeader extends RenderSliverPersiste
     final double maxExtent = this.maxExtent;
     final double paintExtent = maxExtent - _effectiveScrollOffset!;
     final double layoutExtent = maxExtent - constraints.scrollOffset;
+    final double clampedPaintExtent = paintExtent.clamp(0.0, constraints.remainingPaintExtent);
+    final EdgeInsets contentInsets = _getInsetsForAxisDirection(
+      clampedPaintExtent + stretchOffset,
```

Comment:
1, On mobile platform, the AppBar consume the top padding(system status bar) and paint it. This cause the Scrollbar mismatch at the initial state.
2, I also saw that the contentInsets was not updated synchronized when user scrolling.  This cause the scrollbar shake and jump. This needs further investigation.

In my opinion, this is a meaningful new feature and would love to work on this feature with you if you don't mind.

Reply:
> 1, On mobile platform, the AppBar consume the top padding(system status bar) and paint it. This cause the Scrollbar mismatch at the initial state.

Oh really? I thought since the app bar paints into that status bar space, the paintExtent would properly represent it in the reported insets.

> 2, I also saw that the contentInsets was not updated synchronized when user scrolling. This cause the scrollbar shake and jump. This needs further investigation.

Yeah exactly, I have been chasing this for a while trying to figure it out. I thought maybe we need to update the scrollbar painter more often, or maybe not listen to scroll metrics notifications while scroll notification are happening... I have not quote cracked it yet.

> In my opinion, this is a meaningful new feature and would love to work on this feature with you if you don't mind.

I would love that too! I always enjoy collaborating with you. Thank you!